pkg/controller/cronjob/utils.go

```diff
@@ -321,3 +321,16 @@ func adoptJobs(sj *batchv2alpha1.CronJob, js []batchv1.Job, jc jobControlInterfa
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+func getCurrentTimeInZone(sj *batchv2alpha1.CronJob) (time.Time, error) {
+	if sj.Spec.Timezone == nil {
+		return time.Now(), nil
+	}
+
+	loc, err := time.LoadLocation(*sj.Spec.Timezone)
+	if err != nil {
+		return time.Now(), err
```

Comment:
Return nil with error, no time. This may cause some errors slip through. When you return nil, you'll get a nasty nil pointer error which will allow tracking what actually went wrong.